src/Components/test/testassets/BasicTestApp/SelectVariantsComponent.razor

```diff
@@ -20,14 +20,52 @@
     }
 </select>
 
+<select id="select-cars" name="cars" @onchange="SelectedCarsChanged" multiple>
+    <option value="volvo">Volvo</option>
+    <option value="saab">Saab</option>
+    <option value="opel">Opel</option>
+    <option value="audi">Audi</option>
+</select>
+
+<span id="selected-cars-label">@string.Join(", ", SelectedCars)</span>
+
+<select id="select-cities" name="cities" @bind="SelectedCities" multiple>
+    <option value="sf">San Francisco</option>
+    <option value="la">Los Angeles</option>
+    <option value="pdx">Portland</option>
+    <option value="sea">Seattle</option>
+</select>
+
+<span id="selected-cities-label">@string.Join(", ", SelectedCities)</span>
+
 <button @onclick="ToggleShowAdditionalOption" id="show-additional-option"> Show Additional Option</button>
 
 @code
 {
     public string SelectValue { get; set; } = "B";
+    public string[] SelectedCities { get; set; } = new[] { "sf", "sea" };
+    public string[] SelectedCars { get; set; } = new string[] { };
     public bool ShowAdditionalOption = false;
 
     void ToggleShowAdditionalOption() {
         ShowAdditionalOption = true;
     }
+
+    protected override void OnAfterRender(bool firstRender)
+    {
+        if (firstRender)
+        {
+            return;
+        }
+
+        Console.WriteLine($"Selected cities: {string.Join(", ", SelectedCities)}");
+    }
+
+    void SelectedCarsChanged(ChangeEventArgs e)
+    {
+        var array = e.Value as object[];
```

Comment:
I'm wondering if it really makes sense that we're getting an `object[]` here. The reason that `ChangeEventArgs.Value` is typed as `object` is that it works with type=text (`string`), type=checkbox (`bool`), and maybe others.

However, the only type of element that supports "multiple values" is select, and in this case, it's always going to be a `string[]`, isn't it? Could things be simplified if we had a separate property on `ChangeEventArgs`, i.e., `string[] MultipleValues { get; }`?

Reply:
I think this ties into my other suggestion about `GetChangeEventArgsValue` and possibly producing a `string[]` there.